src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -294,7 +294,9 @@ exprt gdb_value_extractort::get_non_char_pointer_value(
   }
   else
   {
-    return it->second;
+    const symbol_exprt typed_symbol_value = symbol_exprt{
+      to_symbol_expr(it->second).get_identifier(), expr.type().subtype()};
```

Comment:
Well I guess that part of the problem is that we initialise the `values` map before we know the types of the symbol expressions we store there. I only introduced this change because these initial symbol expressions do not have to have the right type. I guess we could get the right type during initialisation from symbol table but that change should probably take place in #4261.

Reply:
Ok, not a big deal, was just wondering.